src/main/java/hudson/plugins/warnings/parser/LinuxKernelOutputParser.java

```diff
@@ -0,0 +1,136 @@
+package hudson.plugins.warnings.parser;
+
+import hudson.Extension;
+import hudson.plugins.analysis.util.model.Priority;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * A Parser for Linux Kernel Output detecting WARN() and BUGS()
+ *
+ * @author Benedikt Spranger
+ */
+@Extension
+public class LinuxKernelOutputParser extends RegexpDocumentParser {
```

Comment:
Note that RegexpDocumentParser is very slow since this parser reads the whole log into memory and matches the regexp on this string. In your case it might be smarter to match for the "------------\\[ cut here \\]------------" text and then read the lines until the end trace message.